game/world/objects/interactive.cpp

```diff
@@ -328,39 +334,58 @@ void Interactive::implTick(Pos& p, uint64_t /*dt*/) {
   }
 
 void Interactive::nextState(Npc& npc, MobsiAction act) {
```

Comment:
> Also logic is basically duplication of implTick.

We had this discussion [here](https://github.com/Try/OpenGothic/pull/325#discussion_r996470471). There has to be a way to determine if player is pressing up or down key if a new state is reached. Otherwise you can't use `implTick` and have to copy logic which is called only if a button is pressed.

> Also, which routine AI suppose use and why?

Ai could rely on `implTick` because there's no keyboard interaction, but normal quitting at top has to be handled explicitly. So it uses `nextState` because it's already handled there.

Reply:
Main issues with this design:
1. duplication of interaction-logic. Both `tick` and `nextState` are doing same job.
2. someone who is not aware of this problem, would assume that `nextState` can be used outside of context